src/shogun/classifier/NearestCentroid.cpp

```diff
@@ -54,51 +53,46 @@ namespace shogun{
 		{
 			data = distance->get_lhs();
 		}
-		int32_t num_vectors = data->get_num_vectors();
-		int32_t num_classes = ((CMulticlassLabels*) m_labels)->get_num_classes();
-		int32_t num_feats = ((CDenseFeatures<float64_t>*) data)->get_num_features();
-		SGMatrix<float64_t> centroids(num_feats,num_classes);
+
+		auto* multiclass_labels = m_labels->as<CMulticlassLabels>();
+		auto* dense_data = data->as<CDenseFeatures<float64_t>>();
+
+		int32_t num_classes = multiclass_labels->get_num_classes();
+		int32_t num_feats = dense_data->get_num_features();
+
+		SGMatrix<float64_t> centroids(num_feats, num_classes);
```

Comment:
doesnt the ctor init with zeros these days?

Reply:
I will check it

Reply:
Both only use SG_CALLOC, which I don't think it does.

Reply:
http://www.cplusplus.com/reference/cstdlib/calloc/

Reply:
I didn't think any C function would be so kind to initialize allocated memory!

Reply:
hehe :)